test/test_utils/sagemaker.py

```diff
@@ -0,0 +1,236 @@
+    else:
+        integration_path = os.path.join("integration", sagemaker_test_type)
+
+    # Conditions for modifying tensorflow SageMaker pytest commands
+    if framework == "tensorflow" and sagemaker_test_type == SAGEMAKER_REMOTE_TEST_TYPE:
+        if job_type == "inference":
+            aws_id_arg = "--registry"
+            docker_base_arg = "--repo"
+            integration_path = os.path.join(integration_path, "test_tfs.py")
+            instance_type_arg = "--instance-types"
+
+    if framework == "tensorflow" and job_type == 'training':
+        aws_id_arg = "--account-id"
+
+    test_report = os.path.join(os.getcwd(), "test", f"{tag}.xml")
+    local_test_report = os.path.join(UBUNTU_HOME_DIR, "test", f"{job_type}_{tag}_sm_local.xml")
+    is_py3 = " python3 -m " if "py3" in image else ""
+
+    remote_pytest_cmd = (f"pytest {integration_path} --region {region} {docker_base_arg} "
+                         f"{sm_remote_docker_base_name} --tag {tag} {aws_id_arg} {account_id} "
+                         f"{instance_type_arg} {instance_type} --junitxml {test_report}")
+
+    local_pytest_cmd = (f"{is_py3} pytest -v {integration_path} {docker_base_arg} "
+                        f"{sm_local_docker_base_name} --tag {tag} --framework-version {framework_version} "
+                        f"--processor {processor} {aws_id_arg} {account_id} --junitxml {local_test_report}")
+
+    if framework == "tensorflow" and job_type != "inference":
+        local_pytest_cmd = f"{local_pytest_cmd} --py-version {py_version[2]} --region {region}"
+    if framework == "tensorflow" and job_type == "training":
+        path = os.path.join(os.path.dirname(path), f"{framework}{framework_major_version}_training")
+
+
+    return (
+        remote_pytest_cmd if sagemaker_test_type == SAGEMAKER_REMOTE_TEST_TYPE else local_pytest_cmd,
+        path,
+        tag,
+        job_type
+    )
+
+
+def install_custom_python(python_version, ec2_conn):
+    """
+    Install python 3.6 on Ubuntu 16 AMI.
+    Test files for tensorflow inference require python version > 3.6
+    :param python_version:
+    :param ec2_conn:
+    :return:
+    """
+    ec2_conn.run("sudo add-apt-repository ppa:deadsnakes/ppa -y && sudo apt-get update")
+    ec2_conn.run(f"sudo apt-get install python{python_version} -y ")
+    ec2_conn.run(f"wget https://bootstrap.pypa.io/get-pip.py && sudo python{python_version} get-pip.py")
+    ec2_conn.run(f"sudo ln -sf /usr/bin/python3.6 /usr/bin/python3")
+
+
+def install_sm_local_dependencies(framework, job_type, image, ec2_conn):
+    """
+    Install sagemaker local test dependencies
+    :param framework: str
+    :param job_type: str
+    :param image: str
+    :param ec2_conn: Fabric_obj
+    :return: None
+    """
+    # Install custom packages which need to be latest version"
+    is_py3 = " python3 -m" if "py3" in image else ""
+    # To avoid the dpkg lock with apt-daily service if exists
+    sleep(200)
```

Comment:
This is not for the startup. But to install the virtualenv from apt we will get dpkg lock issue since the daily apt-service update immediately after startup